libnavigation-base/src/main/java/com/mapbox/navigation/base/trip/model/roadobject/RoadObjectLocation.kt

```diff
@@ -0,0 +1,50 @@
+package com.mapbox.navigation.base.trip.model.roadobject
+
+import com.mapbox.geojson.Geometry
+
+/**
+ * RoadObjectLocation contains information about the location of the road object of a concrete
+ * flavor/shape (gantry, polygon, line, point etc.) on the road graph.
+ *
+ * @param locationType type of the road object location
+ * @param shape geometry of the road object
+ */
+abstract class RoadObjectLocation(
+    @RoadObjectLocationType.Type val locationType: Int,
+    val shape: Geometry,
```

Comment:
How is this shape different than the `RoadObjectPosition#coordinate`? If it's the same, why is it duplicated?

Reply:
each child of RoadObjectLocation has shape, some children has extra data, for example 
`PointLocation` has `val position: RoadObjectPosition`
`RoadObjectPosition#coordinate` will be equal to `val shape: Geometry`
in that case it's a duplicate, but `RoadObjectPosition` is used in other places, it must contain  `coordinate `
```
class PolygonLocation internal constructor(
    val entries: List<RoadObjectPosition>,
    val exits: List<RoadObjectPosition>,
    shape: Geometry,
) 
```